crates/engine-primitives/src/lib.rs

```diff
@@ -12,23 +12,14 @@ use core::fmt;
 use reth_primitives::ChainSpec;
 
 use reth_payload_primitives::{
-    BuiltPayload, EngineApiMessageVersion, EngineObjectValidationError, PayloadAttributes,
-    PayloadBuilderAttributes, PayloadOrAttributes,
+    BuiltPayload, EngineApiMessageVersion, EngineObjectValidationError, PayloadOrAttributes, PayloadTypes,
 };
 
 use serde::{de::DeserializeOwned, ser::Serialize};
 /// The types that are used by the engine API.
 pub trait EngineTypes:
-    DeserializeOwned + Serialize + fmt::Debug + Unpin + Send + Sync + Clone
+    PayloadTypes + DeserializeOwned + Serialize + fmt::Debug + Unpin + Send + Sync + Clone
```

Comment:
oh I think this is best solution, this does not change the existing behaviour but splits off the payload types

lgtm!